### PR TITLE
KIP-87: NFT Avatar Standard

### PR DESCRIPTION
**Pull request by yeri-lee**
This NFT avatar standard is to provide an interchangeable visualization of NFT across multiple metaverses.

### CODE REVIEW

File: KIPs/kip-87.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+The NFT avatar standard will generate a positive value creation cycle between creators, the NFT marketplace, and the metaverse. Each individual entity will have a common standard to share, maximizing the utility and interoperability of NFTs. With the standard, the value of an item minted by creators will be increased with heightened practicality and utilization. This will boost the activity within NFT marketplaces while the usage of NFT within metaverses also increases. The versatile utility of NFT characters will increase the public desire to purchase and incentivize creators to generate, ultimately vitalizing the creator economy. Likewise, such dynamics will unleash NFT characters in the multi-metaverse. 
+
+## Specification
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY” and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+NFT metadata structured by the official ERC721 metadata standard or the Enjin Metadata suggestions are as follows referring to OpenSea docs.
```
Comment (baerae-colin) [inline KIPs/kip-87.md]: @yeri-lee Can we have a link to the OpenSea Metadata standard?

File: KIPs/kip-87.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+| ----------- | ----------- |
+| name | The name of the item. |
+| description | A description of the item. |
+| external_url | An external URL that will appear below the asset's image. |
+| image | An URL to the image of the item. This can be any type of image, including SVG, PNG, IPFS URL, or paths. The recommended size of an image is a width between 320 and 1080 pixels and an aspect ratio between 1.91:1 and 4:5 inclusive. |
+| attributes | An attribute for the item, which will show up on the NFT marketplace. |
+| avatars | A metaverse platform where the item can be integrated as NFT avatars. | 
+
+
+Under “avatars,” metadata keys for any metaverse platform, such as Zep.us and Another.world, can be added. The value included in the key will define the animation of a character.
```
Comment (baerae-colin) [inline KIPs/kip-87.md]:
```suggestion
Under "avatars" metadata keys for any metaverse platform, such as Zep.us and Another.world, can be added. The value included in the key will define the animation of a character.  
```

File: KIPs/kip-87.md [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+This proposal currently only includes the standard of Zep NFT avatars, but the standard of additional services can be also defined under “avatars”. 
+
+```JSON Schema
+{
+  "zep": {  
+             "image": "IMAGE URL",
+             "frame_width": 48,
+             "frame_height": 64,
+"animations": {...}
```
Comment (baerae-colin) [inline KIPs/kip-87.md]:
```suggestion
  "zep": {  
             "image": "IMAGE URL",
             "frame_width": 48,
             "frame_height": 64,
             "animations": {...}
```

File: KIPs/kip-87.md [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+| zep | The key defined by zep.us |
+| image | A URL to the image of the avatar sprite sheet. (only PNG images are supported) |
+| frame_width | Width of a frame. The original image width must be divided by the frame_width. It must range between a minimum of 1 and a maximum of the image width. The values of frame_width and frame_height can be different. |
+| frame_height | Height of a frame. The original image height must be divided by the frame_height. It must range between a minimum of 1 and a maximum of the image height. This must be a multiple of an original image size, ranging between a minimum of 1 and a maximum of 256. The values of frame_width and frame_height can be different. | 
+| animations | Animation list of the avatar (idle, moving, jumping, etc…) | 
+
+
+The animation of the NFT avatar is defined under “animations.” The 17 motions are defined: directional movement, involving jump, attack and idle, and dancing. The following array of movements was included in the metadata. Below is an example of the definition of “animations”
+
+```JSON Schema...
```
Comment (baerae-colin) [inline KIPs/kip-87.md]:
```suggestion
```json
```

File: KIPs/kip-87.md [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+{
+"animations": {
```
Comment (baerae-colin) [inline KIPs/kip-87.md]:
```suggestion
     "animations": {
```

File: KIPs/kip-87.md [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+| description | A description of the item. |
+| external_url | An external URL that will appear below the asset's image. |
+| image | An URL to the image of the item. This can be any type of image, including SVG, PNG, IPFS URL, or paths. The recommended size of an image is a width between 320 and 1080 pixels and an aspect ratio between 1.91:1 and 4:5 inclusive. |
+| attributes | An attribute for the item, which will show up on the NFT marketplace. |
+| avatars | A metaverse platform where the item can be integrated as NFT avatars. | 
+
+Under "avatars" metadata keys for any metaverse platform, such as Zep.us and Another.world, can be added. The value included in the key will define the animation of a character.  
+
+```json
+...
```
Comment (baerae-colin) [inline KIPs/kip-87.md]:
```suggestion
```

File: KIPs/kip-87.md [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+Each motion is specified with an array of frames, frame_rate, and repeat. 
+
+| Variable | Description |
+| ----------- | ----------- |
+| frames | Frame index of the sprite sheet (zero-based). Each frame has (frame_width X frame_height) image fragment in the sprite sheet image. The frame index starts from 0. The maximum array length is 256. | 
+| frame_rate | frame_rate must be between a minimum of 1 and a maximum of 64. frame_rate represents the number of frames per second. | 
+| repeat | The number of repetitions of the animation. Only two values are supported: -1 and 1. If it is set to -1, it repeats endlessly. If 1, it repeats once. | 
+
+## Backward Compatibility 
+This standard can be fully EIP-721, EIP-1155, KIP-17, and KIP-37 compatible by adding an extension “avatars” attribute in the metadata. This allows developers to easily adopt the standard quickly.
```
Comment (baerae-colin) [inline KIPs/kip-87.md]: @yeri-lee Could you also add the reference link to KIP-17 and KIP-37?

File: KIPs/kip-87.md [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+
+Each property is defined as below.
+
+| Variable | Description |
+| ----------- | ----------- |
+| zep | The key defined by zep.us |
+| image | A URL to the image of the avatar sprite sheet. (only PNG images are supported) |
+| frame_width | Width of a frame. The original image width must be divided by the frame_width. It must range between a minimum of 1 and a maximum of the image width. The values of frame_width and frame_height can be different. |
+| frame_height | Height of a frame. The original image height must be divided by the frame_height. It must range between a minimum of 1 and a maximum of the image height. This must be a multiple of an original image size, ranging between a minimum of 1 and a maximum of 256. The values of frame_width and frame_height can be different. | 
+| animations | Animation list of the avatar (idle, moving, jumping, etc…) |
```
Comment (baerae-colin) [inline KIPs/kip-87.md]:
```suggestion
| animations | Animation list of the avatar (idle, moving, jumping, etc.) | 
```

**Comment by yeri-lee** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by yeri-lee** [comment]
recheck

File: KIPs/kip-87.md [hunk 9]
```diff
@@ -0,0 +1,240 @@
+---
+kip: 87
+title: NFT Avatar in Multi-Metaverse
+author: Yeri Lee <yeriel.lee@krustuniverse.com>, Junghyun Colin Kim <colin.klaytn@krustuniverse.com>, Wonbae Kim <kernys@supercat.co.kr>, Seokrin Sung <seokrin.sung@supercat.co.kr>
```
Comment (baerae-colin) [inline KIPs/kip-87.md]:
```suggestion
author: Yeri Lee (@yeri-lee), Junghyun Colin Kim (@kjhman21), Wonbae Kim (@kernys), Seokrin Sung (@taronsung)
```

**Comment by baerae-colin** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by baerae-colin — APPROVED** [review]
Let this get merged!

**Pull request merged**